cmd/snap-confine/snap-confine.c

```diff
@@ -329,8 +331,12 @@ int main(int argc, char **argv)
 	// Who are we?
 	uid_t real_uid, effective_uid, saved_uid;
 	gid_t real_gid, effective_gid, saved_gid;
-	getresuid(&real_uid, &effective_uid, &saved_uid);
-	getresgid(&real_gid, &effective_gid, &saved_gid);
+	if (getresuid(&real_uid, &effective_uid, &saved_uid) != 0) {
```

Comment:
```suggestion
	if (getresuid(&real_uid, &effective_uid, &saved_uid) < 0) {
```

Reply:
This falls into the category of https://github.com/snapcore/snapd/pull/7026#discussion_r297842904 where I find the '!= 0' clearer here (since 0 is the only measure of success).